components/collator/README.md

```diff
@@ -9,113 +9,194 @@ and as part of the [`icu`](https://docs.rs/icu/latest/icu/) crate. See the latte
-  normalization part of the code is novel to ICU4X, and the contraction code is custom
-  to ICU4X despite being informed by ICU4C.
-* The way input characters are iterated over and resulting collation elements are
-  buffered is novel to ICU4X.
-* ICU4C can iterate backwards but ICU4X cannot. ICU4X keeps a buffer of the two most
-  recent characters for handling prefixes. As of CLDR 40, there were only two kinds
-  of prefixes: a single starter and a starter followed by a kana voicing mark.
-* ICU4C sorts unpaired surrogates in their lexical order. ICU4X operates on Unicode
-  [scalar values](https://unicode.org/glossary/#unicode_scalar_value) (any Unicode
-  code point except high-surrogate and low-surrogate code points), so unpaired
-  surrogates sort as REPLACEMENT CHARACTERs. Therefore, all unpaired
-  surrogates are equal with each other.
-* Skipping over a bit-identical prefix and then going back over "backward-unsafe"
-  characters is currently unimplemented but isn't architecturally precluded.
-* Hangul is handled specially:
-  - Precomposed syllables are checked for as the first step of processing an
-    incoming character.
-  - Individual jamo are lookup up from a linear table instead of a trie. Unlike
-    in ICU4C, this table covers the whole Unicode block whereas in ICU4C it covers
-    only modern jamo for use in decomposing the precomposed syllables. The point
-    is that search collations have a lot of duplicative (across multiple search)
-    collations data for making archaic jamo searchable by modern jamo.
-    Unfortunately, the shareable part isn't currently actually shareable, because
-    the tailored CE32s refer to the expansions table in each collation. To make
-    them truly shareable, the archaic jamo expansions need to become self-contained
-    the way Latin mini expansions in ICU4C are self-contained.
+assert_eq!(collator_l3.compare("a", "b"), Ordering::Less);  // primary
+assert_eq!(collator_l3.compare("as", "às"), Ordering::Less);  // secondary
+assert_eq!(collator_l3.compare("às", "at"), Ordering::Less);
+assert_eq!(collator_l3.compare("ao", "Ao"), Ordering::Less);  // tertiary
+assert_eq!(collator_l3.compare("Ao", "aò"), Ordering::Less);
+assert_eq!(collator_l3.compare("A", "Ⓐ"), Ordering::Less);
+```
 
-    One possible alternative to loading a different table for "search" would be
-    performing the mapping of archaic jamo to the modern approximations as a
-    special preprocessing step for the incoming characters, which would allow
-    the lookup of the resulting modern jamo from the normal root jamo table.
-
-    "searchjl" is even more problematic than "search", since "searchjl" uses
-    prefixes matches with jamo, and currently Hangul is assumed not to participate
-    in prefix or contraction matching.
+### Alternate Handling
+
+Allows alternate handling for certain customized collation orderings, including the option to
+ignore the special handling for the strings of such customizations.
```

Comment:
It would be good to 1) point out what the only presently-available customization does (moves certain space/punctuation differences from primary to quaternary level and, hence, ignores them at the default tertiary level) and to 2) point out that Thai invokes that customization by default.

Reply:
Done.